### DISCUSSION TITLE
Server Side Meeting Recording.

### DISCUSSION THREAD

**Discussion by 1ay1** [igniterealtime/openfire-pade-plugin#283] [Q&A]
Is there any way we can achieve server side meeting recording apart from the front-end recording we have right now.
The goal is save all the meeting recordings at the server.

**Comment by deleolajide** [comment]
Yes. [Using ffmpeg as I did for live streaming to you-tube](https://github.com/igniterealtime/openfire-pade-plugin/discussions/262).  Instead of sending the stream to an RTMP server, it is written directly to disk as FLV or MP4 file.

**Reply by Jenfong** [reply]
1. I just written directly to disk as FLV file :
`final String cmdLine = ffmpeg + " -i - -vcodec copy -f flv " + room + ".flv";`
for example room is 'demo', the first time click the "Record Desktop App" button, then stop it , it can write to the demo.flv correctly , but the second time it seem do not work yet.
2. I try to write to MP4 file ,but it can not play correctly:
`final String cmdLine = ffmpeg + " -i - -vcodec copy -f mp4 " + room + ".mp4";`
error:
```bat
./ffmpeg.exe -i  demo.mp4
[mov,mp4,m4a,3gp,3g2,mj2 @ 00000225e4abb340] moov atom not found
demo.mp4: Invalid data found when processing input
```

**Reply by gjaekel** [reply]
* If it don't work the second time, remember that `ffmpeg` will not overwrite a existing file without option `-y`.
* If the _moov atom_ is missing, the `ffmpeg` process writing the mp4 stream was not gracefully terminated. For _mp4_, some important information is written at the end of the file. The use `-movflags faststart` while recording to my knowledge allow to playback even broken "end-cutted" mp4's.

**Reply by Jenfong** [reply]
yet ,ffmpeg record to mp4 need some params  I'll try to debug it. 
Now I record to multiple flv files with timstamp ,then merge them to mp4 file by room after room destroyed.
`./ffmpeg.exe -i "concat:of_record_1699446349484.flv|of_record_1699446697601.flv" -c copy of_record.mp4`
+ update pade config 
![snap_2023-11-09T11-16-53 300Z](https://github.com/igniterealtime/openfire-pade-plugin/assets/9347991/7034a96f-5ab0-4582-b7aa-e5dcb420ea7e)
+ record to minio server
[
![snap_2023-11-09T11-25-06 142Z](https://github.com/igniterealtime/openfire-pade-plugin/assets/9347991/0779c27e-0478-4c05-9df1-e2b47687a81c)
](url)

**Comment by deleolajide** [comment]
You can test it without modification to create FLV file. Just put a file path name in the RTMP URL and a subfolder name in the stream key

![image](https://user-images.githubusercontent.com/110731/110972553-050a8880-8354-11eb-9948-9d48a5c16412.png)

**Reply by Jenfong** [reply]
has any webhook or api to fetch the diffrent `rtmp url` and `stream key` dynamically by meeting room in `pade server side` .
eg: 
> live stream auto-save in stream server side

ofmeeting room 1  -> push  live stream to `rtmp://dev-local/live1` 
ofmeeting room 2  -> push  live stream to `rtmp://dev-local/live2`

**Reply by deleolajide** [reply]
No. It was really designed for you-tube that uses the same RTMP URL for all streams of the same you-tube account. It should be straightforward to implement a meeting room property that can override the global URL.

**Comment by 1ay1** [comment]
any way to turn it on by default for all the meetings?

**Comment by deleolajide** [comment]
Not that I aware of
